bin/run.sh

```diff
@@ -89,66 +124,70 @@ else
+    echo "Not starting up. $SCITRAN_PERSISTENT_DB_PATH/mongod.lock reports mongo already running."
+    exit 1
+  fi
 
-# Boostrap users
-if [ $BOOTSTRAP_USERS -eq 1 ]; then
+  echo "Installing Python requirements"
+  bin/install.sh
+
+  if [[ -f "$SCITRAN_RUNTIME_PATH/bin/mongod" ]]; then
+      echo "MongoDB is installed"
+  else
+      echo "Installing MongoDB"
+      curl https://fastdl.mongodb.org/osx/mongodb-osx-x86_64-3.0.7.tgz | tar xz -C $SCITRAN_RUNTIME_PATH --strip-components 1
+      echo "MongoDB installed"
+  fi
+
+  # Launch mongod
+  mongod --dbpath $SCITRAN_PERSISTENT_DB_PATH --smallfiles --port $SCITRAN_PERSISTENT_DB_PORT &
+
+  # Boostrap users
+  if [[ $BOOTSTRAP_USERS -eq 1 ]]; then
     echo "Bootstrapping users"
     bin/bootstrap.py users "$SCITRAN_RUNTIME_BOOTSTRAP"
-else
+  else
     echo "Database exists at $SCITRAN_PERSISTENT_PATH/db. Not bootstrapping users."
-fi
+  fi
 
-TESTDATA_URL="https://github.com/scitran/testdata/archive/master.tar.gz"
-TESTDATA_VERSION=$(curl -sLI $TESTDATA_URL | grep ETag | tail -n 1 | cut -f 2 -d '"')
-if [ ! -d "$SCITRAN_PERSISTENT_PATH/testdata" ]; then
+  TESTDATA_URL="https://github.com/scitran/testdata/archive/master.tar.gz"
+  TESTDATA_VERSION=$(curl -sLI $TESTDATA_URL | grep ETag | tail -n 1 | cut -f 2 -d '"')
+  if [[ ! -d "$SCITRAN_PERSISTENT_PATH/testdata" ]]; then
     echo "Downloading testdata to $SCITRAN_PERSISTENT_PATH/testdata"
     mkdir "$SCITRAN_PERSISTENT_PATH/testdata"
     curl -L $TESTDATA_URL | tar xz -C "$SCITRAN_PERSISTENT_PATH/testdata" --strip-components 1
-else
-    if [ "$TESTDATA_VERSION" != "$(cat $SCITRAN_PERSISTENT_PATH/.testdata_version)" ]; then
-        echo "Testdata out of date; downloading"
-        curl -L $TESTDATA_URL | tar xz -C "$SCITRAN_PERSISTENT_PATH/testdata" --strip-components 1
+  else
+    if [[ "$TESTDATA_VERSION" != "$(cat $SCITRAN_PERSISTENT_PATH/.testdata_version)" ]]; then
+      echo "Testdata out of date; downloading"
+      curl -L $TESTDATA_URL | tar xz -C "$SCITRAN_PERSISTENT_PATH/testdata" --strip-components 1
     else
-        echo "Testdata up to date"
+      echo "Testdata up to date"
     fi
-fi
-builtin echo "$TESTDATA_VERSION" > "$SCITRAN_PERSISTENT_PATH/.testdata_version"
+  fi
+  builtin echo "$TESTDATA_VERSION" > "$SCITRAN_PERSISTENT_PATH/.testdata_version"
 
-if [ -f "$SCITRAN_PERSISTENT_DATA_PATH/.bootstrapped" ]; then
+  if [[ -f "$SCITRAN_PERSISTENT_DATA_PATH/.bootstrapped" ]]; then
     echo "Persistence store exists at $SCITRAN_PERSISTENT_PATH/data. Not bootstrapping data. Remove to re-bootstrap."
-else
+  else
     echo "Bootstrapping testdata"
     bin/bootstrap.py data --copy $SCITRAN_PERSISTENT_PATH/testdata
     echo "Bootstrapped testdata"
     touch "$SCITRAN_PERSISTENT_DATA_PATH/.bootstrapped"
-fi
+  fi
 
-
-# Serve API with PasteScript
-TEMP_INI_FILE=$(mktemp -t scitran_api)
-cat << EOF > $TEMP_INI_FILE
+  # Serve API with PasteScript
+  cat << EOF > $PASTER_INI_FILE
```

Comment:
Never mind, I get it. We need to know the filename on the second run for stopping the daemonized process.